modules/geo-layers/src/tile-3d-layer/tile-3d-layer.js

```diff
@@ -64,6 +66,22 @@ export default class Tile3DLayer extends CompositeLayer {
     }
   }
 
+  activateViewport(viewport) {
+    const {activeViewports, lastUpdatedViewports} = this.state;
+    const oldViewport = this.internalState.viewport;
+    this.internalState.viewport = viewport;
+
+    activeViewports[viewport.id] = viewport;
+    const lastViewport = lastUpdatedViewports?.[viewport.id];
+    if (
+      !(oldViewport && viewport.equals(oldViewport)) &&
```

Comment:
Why still compare with the last viewport? This is exactly what's causing infinite updates.

Reply:
Without these conditions test hangs up because infinite updates happens

Reply:
I'm sure that there are no infinite updates with two viewports . I'm checking it in our I3S debug application: https://github.com/visgl/loaders.gl/tree/master/examples/experimental/i3s-17-and-debug
I do it before every commit

Reply:
It doesn't cause infinite update. See video

Reply:
Ok I read this again. Essentially `oladViewport` and `viewport` are never equal in the multi-view case, so this check is redundant. The `lastUpdatedViewports` map act as a generalized version of the old viewport cache, and `lastViewport` should be the same as `oldViewport` in single-view use case. If the test hangs without this, then there is something else wrong.